File: app/src/main/java/kr/mash_up/seoulmaps/SeoulMapApplication.java
package kr.mash_up.seoulmaps;

import android.app.Application;
import android.content.Context;

/**
 * Created by Tak on 2017. 8. 13..
 */

public class SeoulMapApplication extends Application {
    private static Context context;

    @Override
    public void onCreate() {
        super.onCreate();
        context = this;
    }

    public static Context getContext() {
        return context;
    }
}

File: app/src/main/java/kr/mash_up/seoulmaps/SeoulMapApplication.kt
package kr.mash_up.seoulmaps

import android.app.Application
import android.content.Context
import com.squareup.leakcanary.LeakCanary
import com.squareup.leakcanary.RefWatcher

/**
 * Created by Tak on 2017. 8. 13..
 */

internal class SeoulMapApplication : Application() {

    override fun onCreate() {
        super.onCreate()
        context = this
        setupLeakCanary();
    }

    protected fun setupLeakCanary(): RefWatcher? {
        if (LeakCanary.isInAnalyzerProcess(this)) {
            return RefWatcher.DISABLED;
        }
        return LeakCanary.install(this);
    }

    companion object {
        lateinit var context: Context
            private set
    }
}


